src/core/tsi/ssl_transport_security.cc

```diff
@@ -1254,6 +1255,57 @@ static tsi_result ssl_handshaker_result_extract_peer(
   return result;
 }
 
+static tsi_result ssl_handshaker_result_extract_local_peer(
+    const tsi_handshaker_result* self, tsi_peer* local_peer) {
+  tsi_result result = TSI_OK;
+  const unsigned char* alpn_selected = nullptr;
+  unsigned int alpn_selected_len;
+  const tsi_ssl_handshaker_result* impl =
+      reinterpret_cast<const tsi_ssl_handshaker_result*>(self);
+  X509 *local_cert = SSL_get_certificate(impl->ssl);
+  if (local_cert != nullptr) {
+    result = peer_from_x509(local_cert, 1, local_peer);
+//    X509_free(local_cert);
+    if (result != TSI_OK) return result;
```

Comment:
I do not think we should fail-closed here, can we add a debug-log instead of returning an error?